pgmax/fg/graph.py

```diff
@@ -1002,7 +1022,7 @@ def get_beliefs(bp_arrays: BPArrays) -> Any:
         )
         return beliefs
 
-    return run_bp, get_bp_state, get_beliefs
+    return bp_arrays_init, bp_arrays_update, get_bp_state, get_beliefs
```

Comment:
Instead of returning a tuple, I suggest we bundle these into a dataclass (similar to https://github.com/deepmind/dm-haiku/blob/6f2492db3f06966adff4bb4ce0da3204c422c8b0/haiku/_src/transform.py#L40 but with dataclass not named tuple) with 4 callable attributes. Then the return type would make more sense/is easier to understand and the documentation can be done properly inside the new dataclass

Reply:
We can also change `get_bp_state` to `bp_arrays_to_bp_state` or something similar to make the function easier to understand

Reply:
In addition, here we should return a dataclass bundling 5 functions: the 4 existing ones (`update_bp_arrays`, `run_bp`, `bp_arrays_to_bp_state`, `get_beliefs`), and an additional `init_bp_arrays` (to be consistent having verb first is better) which can be defined as `functools.partial(update_bp_arrays, bp_arrays=None)`.